### ISSUE TITLE
[RangeDatePicker]: labeled component does not have programmatically determinable name for announcement

### ISSUE THREAD

**Issue by mari-shakhanava** [epam/UUI#2376]
### Description
The programmatically determinable name for announcement is missing in the Labeled RangeDatePicker.

### Precondition
The Screen Reader is turned ON.

### Steps to Reproduce

1. Open the [Demo (Form)](https://uui.epam.com/demo?id=form) or create a RangeDatePicker wrapped by a LabeledInput with the `htmlFor` prop provided.
2. Use the 'T' key to navigate to the RangeDatePicker.

### Actual Result

![form-2](https://github.com/epam/UUI/assets/70166849/82a9a9c8-4d49-426a-a780-5fbd5a298498)

The label name is ignored.

### Expected Result
When the combobox is focused, the visible label should be announced by the screen reader. 

The failed WCAG Checkpoint is: 4.1.2 Name, Role, Value (Level A). For more information, visit: [https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html).

---

**Comment by NatalliaAlieva** [comment]
In the current implementation, we have managed to configure the screen reader announcement only for the label of the "from" field. This is due to the specific setup of the label association with the control element within the component.

<img width="867" alt="Screenshot 2024-09-10 at 10 57 58" src="https://github.com/user-attachments/assets/dc54e58c-0a4d-4a6f-9dd9-1604aee97f09">

**Comment by NatalliaAlieva** [comment]
Released in 5.9.2 ver.
